src/Symfony/Component/Console/Input/ArgvInput.php

```diff
@@ -165,11 +165,25 @@ private function parseArgument(string $token)
         // unexpected argument
         } else {
             $all = $this->definition->getArguments();
+            $symfonyCommandName = null;
+            if (($inputArgument = $all[$key = array_key_first($all)] ?? null) && 'command' === $inputArgument->getName()) {
+                $symfonyCommandName = $this->tokens[0] ?? null;
```

Comment:
The command name is not always `$this->tokens[0]`. `bin/console -v secrets:list` will have `-v` as `$this->tokens[0]`

Reply:
You should use `$this->arguments['command'][0]` instead (when we reach this error, we know that the command argument has already been parsed)

Reply:
Awesome. 
You are correct. I've updated the PR. 

```
❯ bin/console -v secrets:list foobar

In ArgvInput.php line 186:
                                                                   
  [Symfony\Component\Console\Exception\RuntimeException]           
  No arguments expected for "secrets:list" command, got "foobar".  

```